lib/puppet/functions/influxdb/retrieve_token.rb

```diff
@@ -2,32 +2,45 @@
 require 'json'
 
 Puppet::Functions.create_function(:'influxdb::retrieve_token') do
+  dispatch :retrieve_token_file do
+    param 'String', :uri
+    param 'String', :token_name
+    param 'Stdlib::Absolutepath', :admin_token_file
```

Comment:
Isn't there a bugreport somewhere that says you cannot use custom datatypes in deffered functions because the functions will be executed on the puppet agent, but the types aren't synced via pluginsync?

Reply:
@bastelfreak found it, PUP-8600